html/status.html

```diff
@@ -229,22 +351,44 @@
 						signal = "<small>OFFLINE</small>";
 				} else {
 					var sig = parseInt(state['ad_strength'][i] / 2.55);
-					var sig0 = Math.max(0, sig - 10);
-					signal = "<div class='level'>STR <div class='pBar' data-from='" + sig0 + "' data-to='" + sig + "'></div> " + sig + "%</div><br />";
+					
+					//STR progressbar color scaling
+					var str_color;
+					if (sig<=25 && sig>=0) str_color="progress-1";
+					if (sig<=42 && sig>25) str_color="progress-2";
+					if (sig<=60 && sig>42) str_color="progress-3";
+					if (sig<=80 && sig>60) str_color="progress-4";
+					if (sig<=100 && sig>80) str_color="progress-5";					
+
+					signal = "<div class='level'>STR <progress class='pstr " + str_color + "' value='" + sig + "' max='100' />" + sig + "%</div><br />";
 
 					if (ad_type == 1 || ad_type == 3)
 						var snr = parseInt(state['ad_snr'][i] / 2.55);
 					else
 						var snr = parseInt(state['ad_snr'][i] / 0.15);
-					var snr0 = Math.max(0, snr - 10);
-					signal += "<div class='level'>SNR <div class='pBar' data-from='" + snr0 + "' data-to='" + snr + "'></div> " + snr + "%</div><br />";
+						var snrdb = parseInt(state['ad_db'][i]);
+
+						if (snrdb >= 65535) // "65535" is the MAX_DB value
+							signal += "<div class='level'>SNR <progress class='psnr " + snr_color + "' value='" + snr + "' max='100' />" + snr + "%</div><br />";
```

Comment:
Can we not set a value if the SNR is not reported ?

Reply:
You say "SNR" or "dB" ?
The code does this:
- When dB is 65535, then we don't have the value. So it isn't printed.
- The default value (in the code) for all adapters are 65535. So, only if a value is reported it will be printed.
- When no SNR is reported the default value is 0. So this value is printed (with the progress bar at zero value).

The only case not considered here is when the "status.html" page gets the "status.json" and the `ad_db` parameter is not reported. In this case, we can assume that the value will be 0. Therefore a "0.0 dB" will be printed. But this is an internall error, because minisatip will exports the `ad_db`. So then we don't need to worry about this extreme case. Right?